hooks/useMetadataForUI.js

```diff
@@ -0,0 +1,127 @@
+import { host, permission } from "../utils/constants";
+
+export default function useMetadataForUI() {
+  const getMetadataForUI = async (library, id) => {
+    try {
+      switch (library) {
+        case "gb":
+          const gbRes = await fetch(
+            `${host}/getMetadata?option=${"gb"}&id=${id}`
+          );
+          const gbMetadata = await gbRes.json();
+          let {
+            title: gb_title,
+            subtitle: gb_subtitle,
+            authors: gb_authors,
+            publisher: gb_publisher,
+            publishedDate: gb_publishedDate,
+            language: gb_language,
+            pageCount: gb_pageCount,
+            infoLink: gb_infoLink,
+          } = gbMetadata.volumeInfo;
+
+          const gb_authorsFormatted = gb_authors
+            ? gb_authors.join().trim()
+            : "";
+          const gb_commonsMetadata = `{{Book
+|Author=${gb_authorsFormatted}
```

Comment:
Use [Space Mono](https://fonts.google.com/specimen/Space+Mono) font to show this in the editor.